homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,180 @@
+            return RESULT_AUTH_MISSING
+        except UnhandledResponse:
+            LOGGER.debug("Working but unsupported config: %s", config)
+            return RESULT_NOT_SUPPORTED
+        except (OSError):
+            LOGGER.debug("Failing config: %s", config)
+
+    LOGGER.debug("No working config found")
+    return RESULT_NOT_FOUND
+
+
+class SamsungTVConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Samsung TV config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+
+    def __init__(self):
+        """Initialize flow."""
+        self._host = None
+        self._ip = None
+        self._manufacturer = None
+        self._model = None
+        self._name = None
+        self._on_script = None
+        self._port = None
+        self._title = None
+        self._uuid = None
+
+    def _get_entry(self):
+        return self.async_create_entry(
+            title=self._title,
+            data={
+                CONF_HOST: self._host,
+                CONF_ID: self._uuid,
+                CONF_IP_ADDRESS: self._ip,
+                CONF_MANUFACTURER: self._manufacturer,
+                CONF_MODEL: self._model,
+                CONF_NAME: self._name,
+                CONF_ON_ACTION: self._on_script,
+                CONF_PORT: self._port,
+            },
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        self._on_script = user_input.get(CONF_ON_ACTION)
+        self._port = user_input.get(CONF_PORT)
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                _get_ip, user_input[CONF_HOST]
+            )
+
+            await self.async_set_unique_id(ip_address)
+            self._abort_if_unique_id_configured()
+
+            self._host = user_input.get(CONF_HOST)
+            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+            self._title = user_input.get(CONF_NAME)
+
+            result = await self.hass.async_add_executor_job(
+                _try_connect, self._host, self._title, self._port
+            )
+
+            if result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            return self._get_entry()
+
+        return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+    async def async_step_ssdp(self, user_input=None):
+        """Handle a flow initialized by discovery."""
+        host = urlparse(user_input[ATTR_SSDP_LOCATION]).hostname
+        ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+
+        await self.async_set_unique_id(ip_address)
+        self._abort_if_unique_id_configured()
```

Comment:
@MartinHjelmare I would like to update manually created devices with the discovered data. Is this allowed and how to do this?

Reply:
Is this what you want?

https://github.com/home-assistant/home-assistant/blob/fffc5a5fbbefa7f20a8f625514effc4366f77f1f/homeassistant/components/axis/config_flow.py#L150-L156

Reply:
Not exactly. `async_set_unique_id` just checks for in progress flows not the configured entries, but I found a way to do so.